velero-plugins/migimagestream/shared.go

```diff
@@ -29,11 +32,28 @@ func copyImage(src, dest string, sourceCtx, destinationCtx *types.SystemContext)
 	if err != nil {
 		return []byte{}, fmt.Errorf("Invalid destination name %s: %v", dest, err)
 	}
-	manifest, err := copy.Image(context.Background(), policyContext, destRef, srcRef, &copy.Options{
-		SourceCtx:      sourceCtx,
-		DestinationCtx: destinationCtx,
-	})
-	return manifest, err
+
+	// Let's retry the image copy up to 10 times
+	// Each retry will wait 5 seconds longer
+	// Let's log a warning if we encounter `blob unknown to registry`
+	retryWait := 5
+	log.Info(fmt.Sprintf("copying image: %s; will attempt up to 5 times...", src))
+	for i := 0; i < 4; i++ {
+		manifest, err := copy.Image(context.Background(), policyContext, destRef, srcRef, &copy.Options{
+			SourceCtx:      sourceCtx,
+			DestinationCtx: destinationCtx,
+		})
+		if err == nil {
+			return manifest, err
+		}
+		if strings.Contains(err.Error(), "blob unknown to registry") {
+			log.Warn(fmt.Sprintf("encountered `blob unknown to registry error` for image %s", src))
+		}
+		log.Info(fmt.Sprintf("attempt #%v failed, waiting %vs and then retrying", i, retryWait))
```

Comment:
We're retrying on any error, not just "blob unknown to registry". Is that by design? This means we'll retry if the registry is down, among other things. If it's an error that takes a timeout to reach, we'll hit 5 timeouts.

Reply:
I agree this isn't ideal but figured it was the easiest approach. I took this approach since I noticed the Docker CLI does the same and containers/image has a conversation to handle this specifically around the 429 HTTP code: https://github.com/containers/image/pull/703.
They do an exponential backoff for 5 retries it looks like. 

I would prefer to dig more into the specific errors this can return and make determinations on it but would need more time to do that properly.

Reply:
Added a TODO to come back to this